cube/main.go

```diff
@@ -1,80 +1,80 @@
 package main
 
 import (
-	"log"
+	"fmt"
 	"time"
 
 	mgl "github.com/go-gl/mathgl/mgl32"
 	"golang.org/x/mobile/app"
 	"golang.org/x/mobile/event"
 	"golang.org/x/mobile/exp/app/debug"
-	"golang.org/x/mobile/exp/gl/glutil"
 	"golang.org/x/mobile/geom"
 	"golang.org/x/mobile/gl"
 )
 
-var (
+type Shape struct {
+	buf     gl.Buffer
+	texture gl.Texture
```

Comment:
the model matrix should probably be associated with the shape struct, since it's supposed to be the transform specific to the thing you are drawing if you have multiple.